wallet/src/wallet/mod.rs

```diff
@@ -153,6 +153,14 @@ pub enum WalletError {
     CannotMintFixedTokenSupply(Amount, Amount, Amount),
     #[error("Trying to unmint {0:?} but the current supply is {1:?}")]
     CannotUnmintTokenSupply(Amount, Amount),
+    #[error("Cannot freeze an unfreezable token")]
+    CannotFreezeUnfreezableToken,
```

Comment:
I think the word Unfreezable is ambiguous. It can mean two things:

1. A token that on issuance, was set to not be freezable (I would call that NonFreezable)
2. A token that was frozen, but can be unfrozen.

I would recommend disambiguating those cases by using the word NonFreezable where appropriate.